### DISCUSSION TITLE
FNIRSI GC-01: Power consumption

### DISCUSSION THREAD

**Discussion by 1MegaSaxoN** [Gissio/radpro#60] [Q&A] [answered]
hi folks, i have installed Gissio's radpro firmware in my unit, but due to poor battery life i am having, i want to go back to the original firmware, however the downloded 1.5 or the 1.6 files are way to big to flash back on, i cannot find the 48k files, my unit has the apm32 mcu with the j613 tube. please help!

**Comment by alfmck** [comment]
My GC-01 uses CH32 mcu and original firmware files flashes OK. I tried to measure the currency the device uses and was surprised that no matter if the radpro or original firmware loaded, the monitor is in sleeping mode or not, device uses 200 mA.

 So I'm thinking about replacing the battery to more powerful one. The problem is that on the original battery are circuits that seems for controlling charging process.

Coming back to original software isn't a good solution, GC-01 with radpro convert it from cheap 5 minutes detector into 24 hours registrator with the possibility of PC connecting for data logging. You can also use any another tube, programming to it's specifications and then calibrate GC-01. Thank Gissio for Your radpro. I was very unsatisfied with the original soft but now I have what I wanted...

**Reply by 1MegaSaxoN** [reply]
i wonder if anyone has successfully upgraded the battery yet? cuz that would be awesome!

**Comment by Gissio** [comment, marked as answer]
[1MegaSaxoN](https://github.com/1MegaSaxoN):

Regarding your first question. On GC-01's with a WCH processor, power consumption of the original firmware is:

* 100 mA@battery with display on (max brightness). Since the battery is 1100 mAh, it should last for 11 hours in this case.
* 53 mA@battery with display off. It should last for 20 hours in this case.

Rad Pro requires 5 to 10 mA less than that, so it should be slightly better.

`alfmck` reports that the power consumption is 200 mA on GC-01's with a Geehy processor, which should last for 5 hours. It would be interesting to know how this was measured (display on, max brightness?). It would also be interesting to know what power consumption is with the display off.

Regarding your second question. Are you using Windows to upload the file? If you are unable to upload the firmware through the USB drive, in a few days I'll post instructions for flashing the firmware with a USB dongle.

**Reply by 1MegaSaxoN** [reply]
i found the proper firmware files, all good.. even though i flashed it right back to the radpro firmware because the unit was a piece of shit with the original firmware... lol going to change the Geiger miller tube next to a j321 form the stock j613 junk that is in it now.. i am still interested in knowing if anyone has been able to install a better battery?

**Reply by Gissio** [reply]
According to my knowledge, it is just a matter of replacing the large battery with a 3.7 V Li-Ion battery of more capacity.

There is also another way of reducing power consumption, but this requires help from volunteers who have some experience in electronics. In Rad Pro, you can set up custom HV profiles to lower power consumption, at the expense of more fluctuation in the high voltage (and thus measurement error). It would be a matter of reducing the PWM frequency and finding the PWM duty cycle that produces the same voltage as the original firmware (440 V). To do this, you *NEED TO MEASURE* the high voltage at the tube. Instructions for doing this are published here: ~https://github.com/Gissio/radpro/blob/main/docs/users.md#hv-profiles~ https://github.com/Gissio/radpro/blob/main/docs/manual.md#hv-profiles

I am *not responsible* if you break your device from overvoltage or overcurrent!

**Reply by dc1rdb** [reply]
Here are some current consumption measurements (median values, because they obviously fluctuate a little) of my GC-01 with Geehy processor, J321 tube and RadPro 2.0rc2. 

1) HV profile factory default
Display high 80 mA
Display med 70 mA
Display sleep 50 mA

2) HV profile custom: 2,5 kHz / 10 %
Display high 60 mA
Display med 50 mA
Display sleep 30 mA

**Reply by Gissio** [reply]
Seems to match the other report for devices with a WCH processor. Thanks for your efforts!

It's also quite interesting that you measured the power consumption with 10 % PWM duty cycle @ 2.5 kHz! Does the tube tick at that setting? Were you able to measure the high voltage at the tube? If it is around 400 V, I can add an "Energy saving" HV profile with these settings.

It would also be necessary for someone with a WCH processor to find the optimum power saving profile: the WCH and Geehy boards are different.

**Reply by dc1rdb** [reply]
Regarding the HV settings, I am still experimenting to find the perfect setting for me. I am measuring the voltage across the geiger tube with a Fluke 80K-40 HV probe.

Since the HV supply of the GC-01 is unregulated (no feedback loop), the key is not to exceed the voltage at low count rates, but have enough reserve for measuring higher activites. So it is always a compromise, depending on the use case.

10% @ 2.5 kHz actually results in overvoltage at my unit. Currently, I'm at 1.5 % @ 5 kHz, which on my unit results in around 440 V at background count rates and decreases to 400 V at activities around 5 uSv/h.

Please keep in mind that there seem to be quite a few different hardware revisions out on the market, so your voltages may differ from mine.

**Reply by Gissio** [reply]
You are absolutely right.

But the idea of an "Energy saving" HV profile to use less power for background level measurements is not a bad one. We actually did this for the Bosean FS-600/FS-1000 and FS2011, and it [performs quite well](https://www.geigerzaehlerforum.de/index.php/topic,1068.135.html). There is, however, one caveat: these devices come with Zener diodes that cap the output to 440 V max, so there is no danger of overvoltage. The GC-01 does not seem to come with these.

**Reply by dc1rdb** [reply]
That is correct, the GC-01 does not have Zener diodes.

**Reply by alfmck** [reply]
I made more accurate HV measurements, using 2.2GOm resistor and analog oscilloscope with 10MOm probe after replacing J614 TO J321. My 440V pairs are: 2.5kHz@1.0%, 5kHz@1.5%, 10 kHz@3%. GC-01 with WCH board, RadPro 2.0rc1 firmware, compiled from source code(zip), submitted in 2.0rc2 release.
10% @ 2.5 kHz really results in overvoltage and should not be used.
Power consumption in MY device is about 200mA, fully charged battery lasted for 6 hours. Display was sleeping most of the time, used for ambient monitoring. I suspect that something is wrong with my device.

**Reply by alfmck** [reply]
Really the power consumption and battery capacity are not so important. You could simply connect it to the power bank when working outside.

**Reply by HFDave** [reply]
> Instructions for doing this are published here: https://github.com/Gissio/radpro/blob/main/docs/users.md#hv-profiles
File not found. Can you update the link please? Thank you.

**Comment by Gissio** [comment]
Could you measure the frequency-duty cycle pairs for the factory-default voltage? From this I could derive an "Optimized" profile which uses less power but provides the same performance as the factory default.

I've already added the 2.5 kHz/1.5 % pair as "energy-saving" profile. It would also be great if you could measure this and ensure it produces 440 V @ background, both on WCH and Geehy variants.

**Reply by alfmck** [reply]
Here are my GC-01 measurements of frequency-duty cycle pairs, after replacing HH614 to J321 and R38-51mOm not placed.
Next week I'll have GC-01 with originally installed J321 and maybe repeat measurements.
[HV settings.xlsx](https://github.com/Gissio/radpro/files/15083084/HV.settings.xlsx)

**Comment by Gerard-Rdam** [comment]
I have a JOY-it JT-RAD01 with Rad Pro 2.0.2 with a
CH32F103R8 processor,
Tube = J321,
R38=5105 and a coil L1=103.

An HV with Factory default setting of 667Volt.
And on the Energy-saving setting of 71 Volt.

I did not have a 1Gohm resistor but 10x 33MOhm=330MOhm and a meter of 10MOhm is about factor 34, so the actual voltage could be about 10volt higher.
The tube starts counting at about 340volt.

I have made a table of this at all frequencies increasing by 1% and higher.

[PWM JOYit Geigermeter.pdf](https://github.com/user-attachments/files/18233984/PWM.JOYit.Geigermeter.pdf)
[Tabel  PWM JOYit Geigermeter.pdf](https://github.com/user-attachments/files/18233985/Tabel.PWM.JOYit.Geigermeter.pdf)

![Tabel PWM JOYit Geigermeter](https://github.com/user-attachments/assets/58037ede-6926-4b40-b510-ed199045935b)

**Reply by Gerard-Rdam** [reply]
I have glued a large hand of an alarm clock with Radium paint on the back of the meter and this gave a measured value of 2,700uS/h with Factory default setting approx. 667Volt.
Then measured at 400Volt with the frequency of 1.5 – 2.5 – 5 – and 10Khz and had almost the same measured values ​​with a maximum deviation of approx. 0.050uS/h.
Only the background value that was approx. 0.150uS/h with Factory default setting is approx. 0.100uS/h at 400Volt.

What did make a big difference was the UV sensitivity of the tube when I shine a UV flashlight on the tube and then only on the negative side of the tube, the positive side was completely insensitive to UV light at a tube voltage of 700Volt.
At 400Volt the tube (J321) was almost insensitive to UV light.
It also seems that the tube on the negative side is more sensitive to ionizing radiation.
I have not measured current, I do not want to cut the battery wire and I do not have a current clamp.

**Comment by ugopapetti** [comment]
That's a very nice analisys, tò complete It Is necessary to have a
measurement of the adsorbed current ti find the best compromise@, let's
Say, 400v fo the dofferent frequencies. Step 2 do the same for systems with
102 (1mH) inductors.  Unfortunately I can not do any measurement in this
moment, really Sorry to not be able tò contribute...

Il lun 23 dic 2024, 23:55 Gerard-Rdam ***@***.***> ha scritto:

**Comment by Gerard-Rdam** [comment]
Fully charged, my JOY-IT RAD01 is empty after more than 30 hours with the screen largely off.

**Comment by STGitUser** [comment]
My FNIRSI runs about that same time length now at 10kHz, 25%.
It wouldn't detect anything below 10k, 20%
At the Default HV settings it detects 40-60 per minute in an environment where 15-20 is shown on my other detectors. This makes me think the default values are pushing the tube too hard, causing false detections.
R38: 915 (9.1MOhm), L1:103, J321 with WCH chipset.

**Reply by Gerard-Rdam** [reply]
The setting I am using here now is 2.5Khz/22% and works well with approx. 400volt here.
I have a resistor R38=51MΩ in it.
That R38 of 9.1MΩ in yours was that already in it or did you put it in yourself? This would then give lower voltage with the same setting...

**Reply by STGitUser** [reply]
Looking at the PCB layout, FNIRSI seems to keep the component labels oriented with the reference designator on the board. So, it appears R38 is a 915 (9.1M), not a 516 (51M). I think we have the same values.

**Reply by Gerard-Rdam** [reply]
I have another question, is R38 915 = 9.1MΩ or 915 underlined = 915MΩ, I just saw on an SMD resistor code app.

**Reply by Gerard-Rdam** [reply]
![R38 915M Ohm](https://github.com/user-attachments/assets/380ff924-eb21-432c-b84a-11e0d98a5066)

**Comment by Gerard-Rdam** [comment]
Does your meter now also indicate the normal background of approx. 10-20 CPM at the HV value you set yourself?

I have added a photo of the R38.

And I have now replaced my tube with an SBM20U tube, which I notice is more sensitive.
I can now use Losalt (potassium) and the ash from cigarettes that I collect from my colleague at work, which also appears to be slightly radioactive.
![R38 ](https://github.com/user-attachments/assets/a1503fdf-1566-43f2-806a-384726e49ca1)

**Reply by STGitUser** [reply]
> Does your meter now also indicate the normal background of approx. 10-20 CPM at the HV value you set yourself?
> 
Yes 10-20, it runs 40 to 60 on the Default setup which is far too high.

**Comment by STGitUser** [comment]
Oh, indeed yours is 51M ohm!  Here's mine.

![20231106_220821](https://github.com/user-attachments/assets/b34f1f59-9c00-48de-b1e1-c26d53153fa8)

**Reply by Gerard-Rdam** [reply]
R38 is indeed 9.1MΩ this would mean that your tube voltage should be lower than mine here with the same settings. I read on the forum that there are also meters where R38 is not in it at all and that there are people who have removed it I don't know why. It is surprising that the manufacturer is also experimenting with this.

**Reply by STGitUser** [reply]
Agreed it is interesting. Even with the standard FNIRSI software the battery life was usually 20 hours or so.

**Comment by ugopapetti** [comment]
Hi Gerard,
saw the snapshot of the SMD resustir app... It indicates 915 milli Ohm, not
Mega Ohm (m, not M). Therefore most probably its a 51MOhm labelled "516" in
this case, and not "5105" likes in others. Also consider that 915MOhm Is
the valute of the resustir used to measure High Voltages without
influencing and High impedence source, as the Voltage multipliers, that
meand that a 915MOhm resisto Is not influencing (too much) the circuito, i
e useless, and we could esclude that a china-firm placed something useless
in a priduct.

Ugo

Il lun 13 gen 2025, 18:20 Gerard-Rdam ***@***.***> ha scritto:

**Reply by Gerard-Rdam** [reply]
You are absolutely right, I had overlooked the small m.
And it should indeed be 516 but because it is a block letter it can also be read as 915 upside down, the font on my resistor looks more like a natural number and then you can't make a mistake with this.
I learned something today, thanks ;-)

**Reply by STGitUser** [reply]
I do believe the resistor value calculator you referenced is using the wrong symbol.  516 and 915 are both Megohm values.  You couldn't buy a resistor precision trimmed to 915 milliohm if you wanted one and it would be a pointlessly small deviation from 1 ohm.  

This is why smd resistors are moving to the EIA-96 labelling standard to avoid these previously possible labeling ambiguities.

https://www.hobby-hour.com/electronics/smdcalc.php

**Reply by Gerard-Rdam** [reply]
The app I use for this is Electrodoc and gives the value in Mohm and if you check an underlined value then in mOhm.
That the value has to be in Mohm is logical otherwise it would be pure short circuit over the high voltage.
![Screenshot_20250204-184712](https://github.com/user-attachments/assets/60adba4a-9aab-4bc5-b715-a000bef752d7)
![Screenshot_20250204-184828](https://github.com/user-attachments/assets/7fb70c39-e75f-4e46-b3fe-2f9b0cfadd1d)
![Screenshot_20250204-184853](https://github.com/user-attachments/assets/21321ae4-1df0-470b-aa84-6b59d4990fbf)

**Comment by Dobfek** [comment]
Hi! An electrical engineer here: No, it is not 9.1MΩ. It is 516, upside down which means 51MΩ... There are two different SMD resistor markings. The 3 digit one, XXY, and the 4 digit one XXXY. X means the value, Y means the zeros after the value, so both of them(5105 and 519) are 51MΩ.

**Reply by Lanchon** [reply]
yeah the difference is:
- 5105: 51 MOhm, 1% tolerance
- 516: 51 MOhm, 5% tolerance
